terraform/modules/self-service/modules/cognito/cognito.tf

```diff
@@ -71,16 +168,13 @@ resource "aws_cognito_user_pool" "user_pool" {
       "mfa_configuration"
     ]
 
-    prevent_destroy = true
+    prevent_destroy = false
   }
 
   provisioner "local-exec" {
-    command = "aws cognito-idp set-user-pool-mfa-config --user-pool-id ${aws_cognito_user_pool.user_pool.id} --software-token-mfa-configuration Enabled=true --mfa-configuration OPTIONAL"
+    command = "aws cognito-idp set-user-pool-mfa-config --user-pool-id ${aws_cognito_user_pool.user_pool.id} --software-token-mfa-configuration Enabled=true --mfa-configuration ON"
   }
 
-  provisioner "local-exec" {
-    command = "aws cognito-idp update-user-pool --user-pool-id  ${aws_cognito_user_pool.user_pool.id} --admin-create-user-config UnusedAccountValidityDays=1"
-  }
```

Comment:
If you are going to destroy then don't we need this when the resource is re created?

Reply:
If we run this it undoes the MFA configuration ;-)